app/frontend/src/pages/chat/Chat.tsx

```diff
@@ -194,7 +203,8 @@ const Chat = () => {
                     }
                 },
                 // AI Chat Protocol: Client must pass on any session state received from the server
-                session_state: answers.length ? answers[answers.length - 1][1].session_state : null
+                // Set new UUID as session state for the first request
+                session_state: answers.length ? answers[answers.length - 1][1].session_state : crypto.randomUUID()
```

Comment:
Do you think we'd also use this randomUUID for apps with logged in users, or would we use Entra OIDs for them? Am wondering if it'd be better to have a helper function to generate this ID.
And if the history storage was in CosmosDB, would we still generate the UUID on the client? Want to make sure this is forward-compatible if possible.

Reply:
Thank you for pointing that out.
You're right, it seems we should think about this carefully.

We need to assign an ID to each "series of chats" to identify them, regardless of whether the user is logged in or not. We'll call this `sessionId`.

## In the case of IndexedDB
- On the client side (React), `sessionId` does not necessarily have to be a UUID, as long as it is a unique key within the client side.
- There is also no need to exchange `sessionId` with the server side using `session_state` (the current implementation of storing `sessionId` in `session_state` is in anticipation of expansion to CosmosDB).

## In the case of CosmosDB
- It is expected that the design will set the user's Entra OID as the partition key and `sessionId` as the unique key.
- In CosmosDB, `conversationId` does not need to be a UUID, as long as it is unique within the partition.
- As mentioned above, `sessionId` does not need to be a UUID on the client side (React).
- `sessionId` is a key in CosmosDB. Therefore, it is preferable to generate it on the server side rather than on the client side.
- As the name `session_state` suggests, it is natural that the `sessionId` stored here is generated on the server side.
- When chat messages are sent and received, the `sessionId` needs to be exchanged between the server side and the client side. `session_state` can be used for this.

## Fix Plan
The plan is to summarize these points and fix them in the following plan.
- If the request from the client contains `session_state`, it will be used as `sessionId`. If not, a new one will be generated on the server and added to `session_state`.
- `sessionId` does not necessarily need to be UUID, but unless there is a specific reason, there is no major problem with UUID, so we will basically generate it with `uuid.uuid4()`.
- Considering the possibility that UUID may be inconvenient when other storage methods are added in the future, we will create a helper function that generates `sessionId` according to the storage method used.